### ISSUE TITLE
Gross pay and Total deductions displayed on the salary slip include components with do not include in total ticked - v13.0.0-beta.9 (version-13-beta)

### ISSUE THREAD

**Issue by muchai** [frappe/erpnext#24411]
This subsequently affects net pay.

This is what is displayed on the salary slip. Because the figures are ok on the salary register report.

https://github.com/frappe/erpnext/pull/23519

**Comment by aakvatech** [comment]
```
    def set_totals(self):
        self.gross_pay = 0
        if self.salary_slip_based_on_timesheet == 1:
            self.calculate_total_for_salary_slip_based_on_timesheet()
        else:
            self.total_deduction = 0
            if self.earnings:
                for earning in self.earnings:
                    if not earning.do_not_include_in_total:
                        self.gross_pay += flt(earning.amount)
            if self.deductions:
                for deduction in self.deductions:
                    if not deduction.do_not_include_in_total:
                        self.total_deduction += flt(deduction.amount)
            self.net_pay = flt(
                self.gross_pay) - flt(self.total_deduction) - flt(self.total_loan_repayment)
        self.set_base_totals()
```

This resolves it. I think the check about do not include in total was skipped.

**Comment by aakvatech** [comment]
https://github.com/frappe/erpnext/blob/6fb66f94289a3a627bb6689418c8512a319101ea/erpnext/payroll/doctype/salary_slip/salary_slip.py#L1100

**Comment by muchai** [comment]
@AfshanKhan This problem arises again in the latest v13.0.0-beta.14 release.

**Comment by muchai** [comment]
It was fixed in v13.0.0-beta.13

**Comment by muchai** [comment]
A slightly different problem, it doesn't set do_not_include_in_total
